### ISSUE TITLE
Not taint specialization on object instantiation 

### ISSUE THREAD

**Issue by ohader** [vimeo/psalm#8284]
PoC: https://psalm.dev/r/49e01bb19a

```php
// tainted HTML is correct here
$a = new Other($_GET['inject']);
echo $a->value;

// ... but it's a false-positive here
$b = new Other('static');
echo $b->value;
```

* property `value` is filled via constructor
* taint graph does not seem to specialize object instantiation with `new`
* adding `@psalm-taint-specialize` to `class Other {}` declaration would solve the problem
* I'm looking for a way to specialize all `new` instances - in general

**Comment by ohader** [comment]
This is how it looks using `@psalm-taint-specialize` (but still, it's not feasible to manipulate each class on a project).
https://psalm.dev/r/a3bf6ed8c6

**Comment by ohader** [comment]
And this is how it would look like in framework-land, having builder, factories, facades triggering object instantiation. It seems all invocations have to use `@psalm-taint-specialize`.

https://psalm.dev/r/458f429c05
